File: Backend/src/main/java/com/altairis_booking_system/Backend/service/HotelService.java
package com.altairis_booking_system.Backend.service;

import com.altairis_booking_system.Backend.domain.Hotel;
import com.altairis_booking_system.Backend.dto.HotelDTO;

import java.util.List;
import java.util.Optional;

public interface HotelService {

    List<HotelDTO> listadoHoteles();

    Optional<HotelDTO> buscarHotelPorId(long idHotel);

    HotelDTO guardarHotel(HotelDTO hotelDTO);

    Optional<HotelDTO> actualizarHotel(HotelDTO hotelDTO, long idHotel);

    void eliminarHotelPorId(long idHotel);

    HotelDTO convertirADTO(Hotel hotel);


}

File: Frontend/src/app/services/hotel.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Hotel } from '../domain/Hotel';

@Injectable({
  providedIn: 'root'
})
export class HotelService {

  private url: string = 'http://localhost:8080/api/hoteles';

  constructor(private http: HttpClient) { 

  }

  listadoHoteles(): Observable<Hotel[]>{
    return this.http.get<Hotel[]>(this.url);
  }
        
  buscarHotelPorId(id: number): Observable<Hotel>{
    return this.http.get<Hotel>(`${this.url}/${id}`);
  }

  buscarPorNombre(nombre: string): Observable<Hotel[]> {
    return this.http.get<Hotel[]>(`${this.url}/buscar?nombre=${nombre}`);
  }
    
  guardarHotel(hotel: Hotel){
    return this.http.post<Hotel>(this.url, hotel);
  }

  editarHotel(hotel: Hotel) {
    return this.http.put<Hotel>(`${this.url}/${hotel.idHotel}`, hotel);
  }

  eliminarHotelPorId(idHotel: number){
    return this.http.delete(`${this.url}/${idHotel}`);
  }

}
